flang/lib/Optimizer/CodeGen/CodeGen.cpp

```diff
@@ -255,8 +255,10 @@ struct AllocaOpConversion : public fir::FIROpConversion<fir::AllocaOp> {
       mlir::Region *parentRegion = rewriter.getInsertionBlock()->getParent();
       mlir::Block *insertBlock =
           getBlockForAllocaInsert(parentOp, parentRegion);
-      size.getDefiningOp()->moveAfter(insertBlock, insertBlock->begin());
-      rewriter.setInsertionPointAfter(size.getDefiningOp());
+      if (rewriter.getInsertionBlock() != insertBlock) {
```

Comment:
This seems okay, but I think there might be cases where we will leave the allocas inside `stacksave/stackrestore` region (which the initial commit tried to avoid).  I am okay with this change to resolve the LIT tests failures, but we may need to revisit this (and fix the tests, if needed).

Reply:
I agree. We would need to revisit to move allocas outside stacksave/restore regions + fix lit tests as necessary.

Reply:
On second thought, I think we would prefer to fix the remaining tests (5 powerpc tests) instead of making another codegen change. I'll go fix those and create a separate merge request to get rid of those lit test failures.